### ISSUE TITLE
Make `defineStore` and `defineActions` instance methods. Remove `register*Descriptor`.

### ISSUE THREAD

**Issue by MrOrz** [tomchentw/goflux#9]
`registerStoreDescriptor` and `defineActionsDescriptor` are instance methods of `Goflux`, which allows us to have multiple instances of `Goflux`.

If  `defineStore` and `defineActions` are instance methods to `Goflux` instead of static methods, we can still have multiple instances of Goflux, but we can eliminate `registerStoreDescriptor` and `defineActionsDescriptor`.

Consider the following usage:

```js
// myFluxInstance.js
module.exports = new require('goflux')
```

Then we can define actions like:

```js
var goflux = require('./myFluxInstance');
goflux.defineActions(......)
```

`myFluxInstance` is a singleton on server. Since dispatcher is managed by context, and the context is generated per reuqest, we can still do isomorphic this way. However, store data must not depend on goflux instance, and should be stored on context as well.

**Comment by tomchentw** [comment]
It make sense! Great idea. I love how it simplify the API and complexibility of `Goflux`. Bravo!

**Comment by tomchentw** [comment]
Here's my proposal for the new API:

```js
const gofluxInstance = new require("goflux");

gofluxInstance.defineActions("UserActions", require("./actions/UserActions"));
gofluxInstance.defineActions("PostActions", require("./actions/PostActions"));

gofluxInstance.defineStore("UserStore", {
  "USER_SIGNED_IN": "onUserSignedIn",
}, require("./stores/UserStore"));

return gofluxInstance;
```

This allow us to reuse the same factory/class of actions and stores for, let's say, admin apps:

```js
const adminInstance = new require("goflux");

adminInstance.defineActions("AdminUserActions", require("./actions/UserActions"));
adminInstance.defineActions("ManagePostActions", require("./actions/PostActions"));

adminInstance.defineStore("UserStore", {
  "USER_SIGNED_IN": "onAdminUserSignedIn",
}, require("./stores/UserStore"));

return adminInstance;
```

**Comment by MrOrz** [comment]
:+1:
:+1:
:+1:

**Comment by tomchentw** [comment]
Closed by #16
